lib/puma/thread_pool.rb

```diff
@@ -424,6 +424,9 @@ def shutdown(timeout=-1)
 
         # Wait +timeout+ seconds for threads to finish.
         join.call(timeout)
+        if shutdown_debugger && !threads.empty?
+          shutdown_debugger.call("Shutdown timeout exceeded")
+        end
```

Comment:
Unlike the graceful shutdown case where we print before we join on the threads, here we print after joining only if the timeout has exceeded. Once if and before forcing a shutdown and then again if and before forcefully killing threads.